src/Microsoft.Identity.Web/TokenCacheProviders/Session/SessionTokenCacheProviderExtension.cs

```diff
@@ -66,7 +66,16 @@ public static IServiceCollection AddSessionTokenCaches(this IServiceCollection s
 
             services.AddHttpContextAccessor();
             services.AddScoped<IMsalTokenCacheProvider, MsalSessionTokenCacheProvider>();
-            services.TryAddScoped(provider => provider.GetService<IHttpContextAccessor>().HttpContext.Session);
```

Comment:
The fact of having a Scoped service does not work with the fact that for Microsoft Graph SDK, they require a singleton. I'm not sure we can do anything but warn developers that they cannot use a Session cache if they use the Microsoft Graph SDK ?